client/src/hooks/subgraph/useEpochTimestampHappenedIn.ts

```diff
@@ -21,15 +23,24 @@ export default function useEpochTimestampHappenedIn(
   timestampSeconds: string,
 ): UseQueryResult<number> {
   const { subgraphAddress } = env;
+  const { data: currentEpoch } = useCurrentEpoch();
+  const { data: isDecisionWindowOpen } = useIsDecisionWindowOpen();
   const timestampSecondsNumber = parseInt(timestampSeconds, 10);
 
   return useQuery<GetEpochTimestampHappenedInQuery, any, number, any>({
-    enabled: !!timestampSecondsNumber,
+    enabled: window.Cypress
+      ? currentEpoch !== undefined && isDecisionWindowOpen !== undefined
+      : !!timestampSecondsNumber,
     queryFn: async () =>
       request(subgraphAddress, GET_EPOCH_TIMESTAMP_HAPPENED_IN, {
         timestamp: timestampSecondsNumber,
       }),
     queryKey: QUERY_KEYS.epochTimestampHappenedIn(timestampSecondsNumber),
-    select: data => data.epoches[0].epoch,
+    select: data => {
+      if (window.Cypress) {
+        return currentEpoch!;
+      }
+      return data.epoches[0].epoch;
```

Comment:
Why?

Reply:
Timestamp in the App is equal to epoch 0.